schemas/Source.yaml

```diff
@@ -4,7 +4,8 @@
 # See the DPDD for more information about the output: https://lse-163.lsst.io
 funcs:
     sourceId:
-        functor: Index
+        functor: Column
+        args: id
```

Comment:
See the comment on this commit: https://github.com/lsst/pipe_tasks/pull/979/commits/1df49118bb78f876ffcfa569a13a9ee7f211b5b5

The `Index` line didn't work with ArrowAstropy when I tried it, and this treats it just like a normal column.

I'm surprised this file works with `calibrateImage` outputs, since that task is not configured to produce an output catalog with many of the fields that this `Source.yaml` expects. That's why there's a new dedicated `.yaml` file for that task's outputs on another commit.

Reply:
> The `Index` line didn't work with ArrowAstropy when I tried it, and this treats it just like a normal column.

Do you remember what didn't work?  I also expected it not to work, but unless I didn't test what I thought I did, restoring the original didn't cause a failure in `transform[Pre]SourceTable` or `consolidate[Pre]SourceTable` for me.  I'm running a ci_hsc in Jenkins now with this one bit reverted (on DM-46991) so that'll hopefully check what I tried to do locally.

> I'm surprised this file works with calibrateImage outputs, since that task is not configured to produce an output catalog with many of the fields that this Source.yaml expects. 

In my incremental approach I've reconfigured the task to produce all of those fields in DRP (except `calib_detected`).  We've got analysis tools that need some of them, but the pipeline steps we have now are really not amenable to adding the multiple invocations we need of those tools (for initial stars, recalibrated stars, and final sources), so I want to fix that on DM-47320 before trying to review which analysis tools we run when, and I want to do that before trimming down the `calibrateImage` output fields back to [close to] the defaults.

Reply:
> Do you remember what didn't work?

No, I don't remember any of the details at this point. I'm pretty sure one of those listed tasks just failed when I tried to use it with ArrowAstropy inputs. It was probably ci_imsim or ci_hsc where the failure occurred.

>  I've reconfigured the task to produce all of those fields in DRP

Oof, that's what I was afraid of. Good luck. I hope we really can get it back to the defaults, and/or any non-default values are ones that we can look over with AP, too.

Reply:
Ok, I figured it out - without this change, `transformSourceTable` does misbehave, but only by replacing the real source ID column with `range(0, N)`, and it turns out nothing actually raises an exception (even downstream _anywhere_ in the pipeline). A lot of tasks just produce no outputs, because `finalizeCharacterization` doesn't produce any PSFs.